Replace plain exception code with ERROR

oflebbe opened the pull request:
First step to remove unnecessary exceptions from the code

michael-o commented on msktconf.cpp:
```diff
@@ -142,12 +142,12 @@ void create_fake_krb5_conf(msktutil_flags *flags)
+        ERROR("setenv failed");
     }
 #else
     int ret = putenv(strdup((std::string("KRB5_CONFIG=") +  g_config_filename).c_str()));
```
Wo does the  `free` after `strdup` because it uses `malloc` in the background?

michael-o commented:
These exits happen before `g_context` has been initialized?

oflebbe commented:
Updated the branch. Do not use an C preprocessor directive, rather a plain function. removed on duplicate error.

Updated again to remove spurious search and replace error

michael-o commented on msktutil.h:
```diff
@@ -294,6 +294,8 @@ int generate_new_password(msktutil_flags *flags);
 #define VERBOSE(text...) if (g_verbose) { fprintf(stdout, " -- %s: ", __FUNCTION__); fprintf(stdout, ## text); fprintf(stdout, "\n"); }
 
+/* Fatal error */
+void error_exit( const char *text);
```
Leading space.

michael-o commented on msktutil.cpp:
```diff
@@ -41,6 +41,11 @@
 
+/* Fatal error */
+void error_exit( const char *text) {
+    fprintf(stderr, "error_exit: %s: %s\n", text, strerror(errno));
```
Does the user really case that `error_exit` is used here? I think all he needs to see is `ERROR`. Everything else is an implementation detail.

oflebbe replied:
The user may not care what failed. But it helps us to diagnose failures. BTW, the message may be seen even before, but was implementation dependent, AFAIK

michael-o commented on msktutil.cpp:
```diff
@@ -60,7 +65,7 @@ std::string sform(const char* format, ...)
 #endif
     if (result < 0) {
-        throw Exception("vasprintf error");
+        error_exit("vasprintf error");
```
`error` not `failed`?

oflebbe replied:
done

oflebbe commented:
@michael-o regarding "These exits happen before g_context has been initialized?" 
Why does this matter ?

michael-o commented:
@oflebbe `initialize_g_context()` is called as second statement in `main`. Only those changed spots are invoked. Error conditions call `exit` and `destroy_g_context` is never called. In fact, it is not destroyed in current master either. Might this be an issue? It is at least not clean code. Am I wrong?

oflebbe commented:
A kerberos context is an in-memory data structure. Since we invoke error_exit when a fatal error occured, we do not care about process memory of a process going to die instantly. 

Cleaning up the kerberos context will not have any effect to the system at this point in time.

oflebbe commented:
I am not saying you are wrong, but the programm is doing a "Whoaa something terrible happend!" with no option to continue sanely. A clean error message is the way to end. 

If we would have to design a library function, this would be a no-go, since a lib should never call exit() and never eat memory, silently.

oflebbe commented:
Hey this was my first github merge ;-)

michael-o commented:
You should have squashed them first and rebased on top of master. The log doesn't look appealing to the reader now.
